### DISCUSSION TITLE
Open archive progress

### DISCUSSION THREAD

**Discussion by kleuter** [rikyoz/bit7z#238] [Q&A]
Hi!

I've got a RAR archive with ~1million files, 7z displays a nice progress bar during opening of the file, however I wasn't able to find such functionality in the library or does it exist?

Thanks.

**Comment by rikyoz** [comment]
Hi!
Unfortunately, the library doesn't currently support total and progress callbacks during opening of archives. It only supports such callbacks during extraction/test/compression operations.
The reason for this is that those callbacks are not easy to integrate into the current library API.
But I'll definitely try to add them, hopefully in the next v4.1-beta.

**Comment by kleuter** [comment]
Thanks for the answer, looking forward to!
